src/lib/components/Error/ErrorReporter.tsx

```diff
@@ -0,0 +1,23 @@
+import { useAtomValue } from 'jotai/utils'
+import { providerAtom } from 'lib/state/web3'
+import { useEffect } from 'react'
+import { EMPTY } from 'widgets-web3-react/empty'
+
+class IntegrationError extends Error {
+  constructor(message: string) {
+    super(message)
+    this.name = 'INTEGRATION ERROR'
+  }
+}
+
+const missingProviderError = new IntegrationError('Missing provider')
+
+export default function ErrorReporter() {
+  const [connector] = useAtomValue(providerAtom)
+  useEffect(() => {
+    if (connector === EMPTY) {
+      throw missingProviderError
```

Comment:
Do we need to throw on this error? We already have defined UI for when there is no connector.